### ISSUE TITLE
MapStruct support: Add builder Method

### ISSUE THREAD

**Issue by lorenzleutgeb** [kobylynskyi/graphql-java-codegen#148]
I'd like `graphql-java-codegen` generated classes to be compatible with MapStruct using the generated builder to instantiate new objects during mapping. The conditions that need to be fulfilled are outlined in the ['Mapping with Builders' section of the respective Reference Guide](https://mapstruct.org/documentation/stable/reference/html/#mapping-with-builders).

As far as I can see, the only missing link is the following:

> The type has a parameterless public static builder creation method that returns a builder. So for example `Person` has a public static method that returns `PersonBuilder`.

Alluding to that example, `graphql-java-codegen` would have to generate:

```java
public class Person {
    // Getters, setters, all as usual here!

    // THIS METHOD NEEDS TO BE GENERATED
    public static PersonBuilder builder() {
        ...
    }

    public static class PersonBuilder {
        // Everything as usual here as well!
        public Person build() {
            ...
        }
}
```

If you approve the addition of this feature, I am happy to implement and PR it.

**Comment by kobylynskyi** [comment]
This sounds good to me. I would appreciate if you can work on this feature.
Thanks, @lorenzleutgeb!

**Comment by kobylynskyi** [comment]
Released in [1.8.0](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v1.8.0)
